divfree/TPZCompElHCurlNoGrads.cpp

```diff
@@ -0,0 +1,369 @@
+#include "TPZCompElHCurlNoGrads.h"
+
+#include <TPZMaterial.h>
+#include <pzconnect.h>
+#include <pzcmesh.h>
+
+template<class TSHAPE>
+TPZCompElHCurlNoGrads<TSHAPE>::TPZCompElHCurlNoGrads() : TPZCompElHCurlFull<TSHAPE>()
+{
+  this->AdjustConnects();
+}
+
+template<class TSHAPE>
+TPZCompElHCurlNoGrads<TSHAPE>::TPZCompElHCurlNoGrads(
+  TPZCompMesh &mesh, TPZGeoEl *gel, int64_t &index) :
+  TPZCompElHCurlFull<TSHAPE>(mesh,gel,index)
+{
+  this->AdjustConnects();
+}
+
+template<class TSHAPE>
+void TPZCompElHCurlNoGrads<TSHAPE>::AdjustConnects()
+{
+  constexpr auto nNodes = TSHAPE::NCornerNodes;
+  constexpr auto ncon = TSHAPE::NSides - nNodes;
+  for(int icon = 0; icon < ncon; icon++){
+    const int connect = this->MidSideConnectLocId(icon+nNodes);
```

Comment:
Wrong use of MidSideConnectLocId -> its input parameter is  a "side" and it returns a "local" connect index
I suggest you create a local variable name "side" and pass that as parameter "even though" your code is correct

Reply:
```suggestion
    const int side = icon + nNodes;
    const int connect = this->MidSideConnectLocId(side);
```

like this?